harness/determined/pytorch/_pytorch_context.py

```diff
@@ -440,6 +487,10 @@ def step_optimizer(
         optimizer: torch.optim.Optimizer,
         clip_grads: Optional[Callable[[Iterator], None]] = None,
         auto_zero_grads: bool = True,
+        scaler: Optional[Any] = None,
+        # Should be torch.cuda.amp.GradScaler, but:
+        #   * other implementations might be possible
+        #   * requiring this type forces upgrades to PyTorch 1.6+
```

Comment:
Other implementations would probably subclass GradScaler, I assume?  But I think leaving it as Any for pytorch versioning reasons is fine.

Reply:
Yeah there's the versioning issue, but also I'm not aware of any other implementations of GradScaler at all - for all I know the first one would simply implement the same interface, and I don't see anything in PyTorch that would force it to be subclassed.